x-pack/plugins/remote_clusters/public/sections/remote_cluster_form/remote_cluster_form.js

```diff
@@ -42,576 +42,571 @@ const defaultFields = {
+    static defaultProps = {
+      fields: merge({}, defaultFields),
+      disabledFields: {},
+    }
 
-  constructor(props) {
-    super(props);
+    constructor(props) {
+      super(props);
 
-    const { fields, disabledFields } = props;
-    const fieldsState = merge({}, defaultFields, fields);
+      const { fields, disabledFields } = props;
+      const fieldsState = merge({}, defaultFields, fields);
 
-    this.state = {
-      localSeedErrors: [],
-      fields: fieldsState,
-      disabledFields,
-      fieldsErrors: this.getFieldsErrors(fieldsState),
-      areErrorsVisible: false,
-    };
-  }
+      this.state = {
+        localSeedErrors: [],
+        fields: fieldsState,
+        disabledFields,
+        fieldsErrors: this.getFieldsErrors(fieldsState),
+        areErrorsVisible: false,
+      };
+    }
 
-  getFieldsErrors(fields) {
-    const { name, seeds } = fields;
+    getFieldsErrors(fields) {
+      const { name, seeds } = fields;
 
-    const errors = {};
+      const errors = {};
 
-    if (!name || !name.trim()) {
-      errors.name = (
-        <FormattedMessage
-          id="xpack.remoteClusters.form.errors.nameMissing"
-          defaultMessage="Name is required."
-        />
-      );
-    } else if (name.match(/[^a-zA-Z\d\-_]/)) {
-      errors.name = (
-        <FormattedMessage
-          id="xpack.remoteClusters.form.errors.illegalCharacters"
-          defaultMessage="Name can only contain letters, numbers, underscores, and dashes."
-        />
-      );
-    }
+      if (!name || !name.trim()) {
+        errors.name = (
+          <FormattedMessage
+            id="xpack.remoteClusters.form.errors.nameMissing"
+            defaultMessage="Name is required."
+          />
+        );
+      } else if (name.match(/[^a-zA-Z\d\-_]/)) {
+        errors.name = (
+          <FormattedMessage
+            id="xpack.remoteClusters.form.errors.illegalCharacters"
+            defaultMessage="Name can only contain letters, numbers, underscores, and dashes."
+          />
+        );
+      }
 
-    if (!seeds.some(seed => Boolean(seed.trim()))) {
-      errors.seeds = (
-        <FormattedMessage
-          id="xpack.remoteClusters.form.errors.seedMissing"
-          defaultMessage="At least one seed is required."
-        />
-      );
+      if (!seeds.some(seed => Boolean(seed.trim()))) {
+        errors.seeds = (
+          <FormattedMessage
+            id="xpack.remoteClusters.form.errors.seedMissing"
+            defaultMessage="At least one seed is required."
+          />
+        );
+      }
+
+      return errors;
     }
 
-    return errors;
-  }
+    onFieldsChange = (changedFields) => {
+      const { fields: prevFields } = this.state;
 
-  onFieldsChange = (changedFields) => {
-    const { fields: prevFields } = this.state;
+      const newFields = {
+        ...prevFields,
+        ...changedFields,
+      };
 
-    const newFields = {
-      ...prevFields,
-      ...changedFields,
+      this.setState({
```

Comment:
wouldn't it be better here to use `this.setState(prevState => ({}))`?

Reply:
Good catch! This is being fixed in a separate PR.